rust/src/dataset/scanner.rs

```diff
@@ -137,8 +137,15 @@ impl Scanner {
     /// The schema of the output, a.k.a, projection schema.
     pub fn schema(&self) -> SchemaRef {
         if self.nearest.as_ref().is_some() {
+            let q = self.nearest.as_ref().unwrap();
+            let column: ArrowField = self
+                .dataset
+                .schema()
+                .field(q.column.as_str())
+                .unwrap()
```

Comment:
Is this an `Option`? do a `ok_or_else()`?

Reply:
done. also had to change return type of schema to Result<SchemaRef>